torch/csrc/autograd/function.h

```diff
@@ -146,6 +146,10 @@ struct TORCH_API Node : std::enable_shared_from_this<Node> {
   /// Evaluates the function on the given inputs and returns the result of the
   /// function call.
   variable_list operator()(variable_list&& inputs) {
+    if (AnomalyMode::is_enabled()) {
```

Comment:
Alternatively, this could be implemented as a grad_fn's hook: https://gist.github.com/soulitzer/28140cc4cd7d26828ff7f07b1235d9f5

Reply:
That is a much more flexible solution, so I think we want to go with hooks here.

Reply:
OK, will do.